### ISSUE TITLE
ttsc --watch not working when upgrade typescript-transform-paths to 2.0.3

### ISSUE THREAD

**Issue by Val-istar-Guo** [LeDDGroup/typescript-transform-paths#78]
* "typescript-transform-paths": "^2.0.3"
* "ttypescript": "^1.5.12",
* "typescript": "^4.1.2",
* "typescript-nestjs-swagger-plugin": "^1.1.0",

```
***/node_modules/typescript/lib/typescript.js:14332
 :ts.skipTrivia(sourceFile.text, errorNode.pos);

TypeError: Cannot read property 'text' of undefined
```

Operating procedures:

- `ttsc --watch` 
- change some ts file that be watched
- The error throw

https://github.com/cevek/ttypescript/issues/92

**Comment by nonara** [comment]
Thanks for filing!

I was working on a similar issue today and was able to work out what's going on.

In your case, the short version is, the parent node isn't being set on our workaround node. There also doesn't seem to be a way that we can do it without messing up the output.

You can read more on that if you're interested in [this issue](https://github.com/LeDDGroup/typescript-transform-paths/issues/77). Ultimately, we're going to have to wait on TS to fix the issue that we've been working around.

The [TS issue](https://github.com/microsoft/TypeScript/issues/40603#event-3815630846) is in the roster for 4.2.0, and I'm keeping an eye on it.

I'm going to leave this issue open also, however, so that when we have a fix, I'll be sure to test your watch-mode case specifically.

**Comment by jonathanj** [comment]
@nonara Just checking if this is different from #72? You commented on a few days ago indicating that the issue there is fixed, but my understanding of the two issues is that they are the same.

I'm experiencing the same issue above trying to use `typescript-transform-paths@2.0.4` on `typescript@4.1.2` with `ttypescript@1.5.12`, except I'm not specifying `--watch` I'm just trying to compile my TS project to JavaScript + TS declarations for publishing purposes.

```
yarn run v1.22.5
$ /Users/jonathan/Coding/yoco/design-system-react-native/node_modules/.bin/ttsc --declaration --project .
/Users/jonathan/Coding/yoco/design-system-react-native/node_modules/typescript/lib/typescript.js:106219
                throw e;
                ^

TypeError: Cannot read property 'text' of undefined
```

**Comment by jonathanj** [comment]
I tried adding the `afterDeclarations: true` to the transformer options, and now `ttsc` doesn't crash but it also doesn't transform the paths in the compiled JS output, only the paths in the TypeScript declaration files.

```
cat dist/index.js
…
const Grid_1 = __importDefault(require("@/components/layout/Grid"));
```

```
cat dist/index.d.ts
…
import Grid from "./src/components/layout/Grid/index";
```

**Comment by nonara** [comment]
@jonathanj It is different than #72, though somewhat related. It is a duplicate of #77 

In v2.0.3, we were able to alleviate the issue for many cases, however, as these two issues have shown, there are still edge cases where it breaks.

Background:
- [Detail on the TS bug & our workaround](https://github.com/LeDDGroup/typescript-transform-paths/pull/76)
- [Detail on what's still breaking & where we're at](https://github.com/LeDDGroup/typescript-transform-paths/issues/77#issuecomment-731891246)

It appears at the moment that our only option is to wait for a fix in TS. I've talked to some MSFT folks and gotten some detail on where to look, so I'll be looking at the TS source today to see if I can help expedite a fix for the TS issue. Once we can drop the workarounds, all of these issues will be settled.
